apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js

```diff
@@ -0,0 +1,1013 @@
+
+  suite('_onAdapterDeviceunpaired > ', function() {
+    var mockAdapter;
+    setup(function() {
+      mockAdapter = {};
+      this.sinon.stub(btContext, '_refreshPairedDevicesInfo');
+    });
+
+    test('_refreshPairedDevicesInfo() should be called with adapter ',
+    function() {
+      btContext._onAdapterDeviceunpaired(mockAdapter);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.called);
+      assert.isTrue(btContext._refreshPairedDevicesInfo.calledWith(
+        mockAdapter));
+    });
+  });
+
+  suite('_onDefaultAdapterChanged > ', function() {
+    var newAdapter, oldAdapter;
+    setup(function() {
+      newAdapter = {};
+      oldAdapter = {};
+      this.sinon.stub(btContext, '_defaultAdapter');
+      this.sinon.stub(btContext, '_unwatchDefaultAdapterOnattributechanged');
+      this.sinon.stub(btContext, '_unwatchDefaultAdapterOndeviceunpaired');
+      this.sinon.stub(btContext, '_unwatchDefaultAdapterOndevicepaired');
+      this.sinon.stub(btContext, '_initProperties');
+      this.sinon.stub(btContext, '_watchDefaultAdapterOnattributechanged');
+      this.sinon.stub(btContext, '_watchDefaultAdapterOndevicepaired');
+      this.sinon.stub(btContext, '_watchDefaultAdapterOndeviceunpaired');
+      this.sinon.stub(btContext, '_resetProperties');
+    });
+
+    test('default adapter is chaned with new/old adapter ', function() {
+      btContext._onDefaultAdapterChanged(newAdapter, oldAdapter);
+      // test default adapter updated
+      assert.equal(btContext._defaultAdapter, newAdapter);
+      // test unwatch events
+      assert.isTrue(btContext._unwatchDefaultAdapterOnattributechanged.called);
+      assert.isTrue(
+        btContext._unwatchDefaultAdapterOnattributechanged.calledWith(
+          oldAdapter));
+      assert.isTrue(btContext._unwatchDefaultAdapterOndeviceunpaired.called);
+      assert.isTrue(
+        btContext._unwatchDefaultAdapterOndeviceunpaired.calledWith(
+          oldAdapter));
+      assert.isTrue(btContext._unwatchDefaultAdapterOndevicepaired.called);
+      assert.isTrue(
+        btContext._unwatchDefaultAdapterOndevicepaired.calledWith(oldAdapter));
+      // test init porperties
+      assert.isTrue(btContext._initProperties.called);
+      assert.isTrue(btContext._initProperties.calledWith(newAdapter));
+      // test watch events
+      assert.isTrue(btContext._watchDefaultAdapterOnattributechanged.called);
+      assert.isTrue(
+        btContext._watchDefaultAdapterOnattributechanged.calledWith(
+          newAdapter));
+      assert.isTrue(btContext._watchDefaultAdapterOndevicepaired.called);
+      assert.isTrue(
+        btContext._watchDefaultAdapterOndevicepaired.calledWith(newAdapter));
+      assert.isTrue(btContext._watchDefaultAdapterOndeviceunpaired.called);
+      assert.isTrue(
+        btContext._watchDefaultAdapterOndeviceunpaired.calledWith(newAdapter));
+      // test reset properties
+      assert.isFalse(btContext._resetProperties.called);
+    });
```

Comment:
Please add a case ensuring `_resetProperties` is called when `newAdapter` is null.